schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/update-fixed-info.xsl

```diff
@@ -141,7 +146,6 @@
       <!-- Preserve date order -->
       <xsl:for-each select="mdb:dateInfo">
```

Comment:
This code doesn't seem right, it should  **only**  loop over the existing `mdb:dateInfo` that are in the metadata to process, but if I'm not misunderstanding the change you want to process also the creation and revision dates added in the previous lines?

I don't think that works.

Reply:
no, above we add creation or revision if not existing.
here 
* if revision we update the date
* if empty creation date we skip as it was added before with current time
* else we copy.

Reply:
I got confused with the comment `remove empty creation date, added before if emtpy.`, I thought you were trying to process the creation / revision dates added before also. As this code can add an empty value if `not($isCreationDateAvailable)` is `true` and `/root/env/createDate = ''`:

https://github.com/geonetwork/core-geonetwork/pull/6968/files#diff-640f6f2da2cd955314b15fa9f7982759c5ac129528d58e90c6b4d56d6ca3e84fR118-R120

Reply:
Can be updated the text: `remove empty creation date, added before if emtpy` to be more clear?

To me it's quite confusing what `before` refers to, it seems something added previously in the xsl, but doesn't seem the purpose of the comment.